[Bug]: Embedded cannot run with default configurations

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.2.2
- Deployment mode(standalone or cluster): docker
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```


### Current Behavior

I run with:
```
docker run --name=milvus --rm -p9091:9091 -p19530:19530  milvusdb/milvus:v2.2.2 milvus run embedded
```

```log
[2023/02/03 05:31:31.766 +00:00] [INFO] [paramtable/base_table.go:143] ["cannot find etcd.endpoints"]
[2023/02/03 05:31:31.766 +00:00] [INFO] [logutil/logutil.go:165] ["Log directory"] [configDir=]
[2023/02/03 05:31:31.766 +00:00] [INFO] [logutil/logutil.go:166] ["Set log file to "] [path=]
[2023/02/03 05:31:31.766 +00:00] [INFO] [roles/roles.go:192] ["starting running Milvus components"]
[2023/02/03 05:31:31.766 +00:00] [INFO] [roles/roles.go:134] ["Enable Jemalloc"] ["Jemalloc Path"=/milvus/lib/libjemalloc.so]
[2023/02/03 05:31:31.767 +00:00] [INFO] [paramtable/base_table.go:143] ["cannot find etcd.endpoints"]
panic: key not found: etcd.endpoints

goroutine 1 [running]:
github.com/milvus-io/milvus/internal/util/paramtable.(*EtcdConfig).initEndpoints(0x5699a00)
	/go/src/github.com/milvus-io/milvus/internal/util/paramtable/service_param.go:139 +0x97
github.com/milvus-io/milvus/internal/util/paramtable.(*EtcdConfig).LoadCfgToMemory(0x5699a00)
	/go/src/github.com/milvus-io/milvus/internal/util/paramtable/service_param.go:106 +0x4c
github.com/milvus-io/milvus/internal/util/paramtable.(*EtcdConfig).init(...)
	/go/src/github.com/milvus-io/milvus/internal/util/paramtable/service_param.go:97
github.com/milvus-io/milvus/internal/util/paramtable.(*ServiceParam).Init(0x5699900)
	/go/src/github.com/milvus-io/milvus/internal/util/paramtable/service_param.go:61 +0xc6
github.com/milvus-io/milvus/internal/util/paramtable.(*ComponentParam).Init(0x5699900)
	/go/src/github.com/milvus-io/milvus/internal/util/paramtable/component_param.go:80 +0x25
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc00061be50, 0x1, {0x0, 0x0})
	/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:201 +0x35b
github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc000a1e420, {0xc00011c060?, 0x3, 0x3}, 0xc000506d80)
	/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:120 +0x733
github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc00011c060?, 0x3, 0x3})
	/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e
main.main()
	/go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e
```

### Expected Behavior

I used to run with v2.1.2:
```
docker run --name=milvus --rm -p9091:9091 -p19530:19530  milvusdb/milvus:v2.1.2 milvus run embedded
[2023/02/03 05:30:43.484 +00:00] [INFO] [paramtable/base_table.go:459] ["Pulsar config"] ["pulsar url"=pulsar://localhost:6650] ["pulsar web url"=http://localhost:80]
2023-02-03 05:30:44,171 INFO [default] [KNOWHERE][SetBlasThreshold][milvus] Set faiss::distance_compute_blas_threshold to 16384
2023-02-03 05:30:44,171 INFO [default] [KNOWHERE][SetEarlyStopThreshold][milvus] Set faiss::early_stop_threshold to 0
2023-02-03 05:30:44,171 INFO [default] [KNOWHERE][SetStatisticsLevel][milvus] Set knowhere::STATISTICS_LEVEL to 0
---Milvus Proxy successfully initialized and ready to serve!---
```

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

--- a/cmd/milvus/run.go
+++ b/cmd/milvus/run.go
@@ -12,6 +12,7 @@ import (
 	"github.com/milvus-io/milvus/internal/log"
 	"github.com/milvus-io/milvus/internal/util/hardware"
 	"github.com/milvus-io/milvus/internal/util/metricsinfo"
+	"github.com/milvus-io/milvus/internal/util/paramtable"
 	"github.com/milvus-io/milvus/internal/util/typeutil"
 )
 
@@ -90,6 +91,12 @@ func (c *run) execute(args []string, flags *flag.FlagSet) {
 	default:
 		fmt.Fprintf(os.Stderr, "Unknown server type = %s\n%s", c.serverType, c.getHelp())
 		os.Exit(-1)
+	}
+
+	// setup config for embedded milvus
+	if c.serverType == typeutil.EmbeddedRole {
+		var params paramtable.BaseTable
+		params.GlobalInitWithYaml("embedded-milvus.yaml")
 	}
 
 	runtimeDir := createRuntimeDir(c.serverType)

--- a/internal/util/paramtable/base_table.go
+++ b/internal/util/paramtable/base_table.go
@@ -17,6 +17,7 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	config "github.com/milvus-io/milvus/internal/config"
@@ -57,7 +58,8 @@ var defaultYaml = DefaultMilvusYaml
 
 // BaseTable the basics of paramtable
 type BaseTable struct {
-	mgr *config.Manager
+	once sync.Once
+	mgr  *config.Manager
 
 	configDir string
 	YamlFile  string
@@ -74,6 +76,16 @@ func NewBaseTableFromYamlOnly(yaml string) *BaseTable {
 	}))
 	gp := &BaseTable{mgr: mgr, YamlFile: yaml}
 	return gp
+}
+
+// GlobalInitWithYaml initializes the param table with the given yaml.
+// We will update the global DefaultYaml variable directly, once and for all.
+// GlobalInitWithYaml shall be called at the very beginning before initiating the base table.
+// GlobalInitWithYaml should be called only in standalone and embedded Milvus.
+func (gp *BaseTable) GlobalInitWithYaml(yaml string) {
+	gp.once.Do(func() {
+		defaultYaml = yaml
+	})
 }
 
 // init initializes the param table.

